build/circle-deps.sh

```diff
@@ -85,3 +107,48 @@ grep -v '^cmd' GLOCKFILE | glock sync -n
 # Recursively invoke this script inside the builder docker container,
 # passing "docker" as the first argument.
 $(dirname $0)/builder.sh $0 docker
+
+# According to
+# https://discuss.circleci.com/t/cache-save-restore-algorithm/759, the
+# cache is only collected from container (shard) 0. We workaround this
+# limitation by copying the build output from shards 1 and 2 over to
+# shard 0. On shard 0 we wait for this remote data to be copied.
+
+gopath0="${GOPATH%%:*}"
+
+if is_shard 2; then
```

Comment:
Yes, though note that we want to run shards 1 and 2 before shard 0. So your choices are `1, 2, 0` or `2, 1, 0`. Why do we want to run `shard 0` last? Because we might not have parallelism turned on.